pkg/apis/uiplugin/v1alpha1/types.go

```diff
@@ -31,20 +32,50 @@ type UIPluginList struct {
 	Items           []UIPlugin `json:"items"`
 }
 
-// +kubebuilder:validation:Enum=Dashboards
+// +kubebuilder:validation:Enum=Dashboards;Logging
 type UIPluginType string
 
 const (
-	// Dashboards deploys the Dashboards Dynamic Plugin for OpenShift Console.
+	// TypeDashboards deploys the Dashboards Dynamic Plugin for OpenShift Console.
 	TypeDashboards UIPluginType = "Dashboards"
+
+	// TypeLogging deploys the Logging View Plugin for OpenShift Console.
+	TypeLogging UIPluginType = "Logging"
 )
 
+// LoggingConfig contains options for configuring the logging console plugin.
+type LoggingConfig struct {
+	// LokiStack points to the LokiStack instance of which logs should be displayed.
+	// It always references a LokiStack in the "openshift-logging" namespace.
+	//
+	// +kubebuilder:validation:Required
+	LokiStack corev1.LocalObjectReference `json:"lokiStack"`
+
+	// LogsLimit is the max number of entries returned for a query.
+	//
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OCP Console Log Limit",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:ocpConsoleLogLimit"}
+	LogsLimit int `json:"logsLimit,omitempty"`
+
+	// Timeout is the max duration before a query timeout
+	//
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OCP Console Query Timeout",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:ocpConsoleTimeout"}
+	// +kubebuilder:validation:Pattern:="^([0-9]+)([smhd]{0,1})$"
+	Timeout string `json:"timeout,omitempty"`
```

Comment:
Do we even need to support a unit suffix (and assume seconds)? If yes, I suppose that it can't be omitted.

Reply:
The timeout here is usually in the order of minutes and I think it's easier (and more "visual") to configure `3m` instead of `180`. The default unit is seconds already.

Reply:
Ack. But hours and days seem a bit overkill?
Also I fail to see in https://github.com/openshift/logging-view-plugin/ where it assumes seconds if the unit's missing.

Reply:
I assumed it used seconds, because when I looked through the source briefly I spotted a function converting it to seconds for the internal representation for the JavaScript backend.

When actually testing this, it became more complex though. I have now added code that actually does what we advertise (choice of seconds and minutes, default of seconds) in the operator and formats the ConfigMap passed to the logging-view-plugin accordingly.